src/main/java/com/karol/kindergartenmanagementsystem/exception/ControllerExceptionHandler.java

```diff
@@ -0,0 +1,34 @@
+package com.karol.kindergartenmanagementsystem.exception;
+
+import com.karol.kindergartenmanagementsystem.http.ApiErrorResponse;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@ControllerAdvice
+public class ControllerExceptionHandler {
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ResponseEntity<ApiErrorResponse> handleValidationErrors(MethodArgumentNotValidException exception) {
+        Map<String, String> errors = new HashMap<>();
+
+        exception.getBindingResult().getAllErrors().forEach((error) -> {
+            String fieldName = ((FieldError) error).getField();
+            String errorMessage = error.getDefaultMessage();
+            errors.put(fieldName, errorMessage);
+        });
```

Comment:
To dasz radę zrobić to bez uprzedniego instancjonowania mapy.
https://www.baeldung.com/java-collectors-tomap